motionplan/tpSpaceRRT.go

```diff
@@ -306,6 +306,36 @@ func (mp *tpSpaceRRTMotionPlanner) rrtBackgroundRunner(
 			rrt.solutionChan <- &rrtSolution{err: fmt.Errorf("TP Space RRT timeout %w", ctx.Err()), maps: rrt.maps}
 			return
 		}
+
+		// Check to see if the startPose is already within some predefined delta of the goalPose
+		if iter == 0 {
+			atGoal := func(startPose, goalPose spatialmath.Pose) bool {
```

Comment:
This is already handled (more accurately) by mp.planOpts.DistanceFunc. Use that instead of this.

Reply:
will do. 
question: we define `DistanceFunc` as `L2InputMetric`
```
InputsL2Distance returns the square of the two-norm between the from and to vectors.
```
does this account for when we are not operating in PositionOnly mode?

Reply:
Ah, you're right, it wouldn't.

However, within the algorithm itself is not an appropriate place to be checking business logic such as Position Only Mode.

I think we need a third way.

Reply:
>However, within the algorithm itself is not an appropriate place to be checking business logic such as Position Only Mode.

It is not immediately obvious to me why you say that this shouldn't live at the algorithm level. Is it because it introduces special handling code/tech-debt? What are your thoughts?


If it does not live on the algorithm level, "it" being the early return from the planner, then "it" must live either before we enter the motionplan package or once we return from it. The main branch currently handles the latter -- the check existing once we exit the planner.